c_src/workitems.cc

```diff
@@ -155,7 +157,18 @@ OpenTask::operator()()
     DbObject * db_ptr;
     rocksdb::DB *db(0);
 
-    rocksdb::Status status = rocksdb::DB::Open(*options, db_name, &db);
+    rocksdb::Status status;
+
+    if(ttl <= 0)
+    {
+        status = rocksdb::DB::Open(*options, db_name, &db);
+    }
+    else
+    {
+         rocksdb::DBWithTTL* db_with_ttl(0);
+         status = rocksdb::DBWithTTL::Open(*options, db_name, &db_with_ttl, ttl);
+         db = db_with_ttl;
+    }
```

Comment:
Please tell us the reason why you've decided to adopt this behavior (if ttl is set to 0 or less than 0 then fallback to the normal rocksdb::DB::Open).

According to the official document at https://github.com/facebook/rocksdb/wiki/Time-to-Live, rocksdb behave as TTL = infinity in such cases.

Without any comment on the code, erocksdb users might expect the code to behave what the official document says.

That  being said, if there are some rational reason to fallback then we'd have to document/comment about it on the comment of function open/4 at src/erocksdb.erl.

Reply:
hmm .. infinity doesn't mean should behave like the default open function ? This was my first thought when I readed the documentation. That in case you open using rocksdb::DBWithTTL::Open having TTL <= 0 it's same stuff like using rocksdb::DB::Open where records are never removed based on any TTL..

Reply:
Yes partly I think so.
Although I've not confirmed yet, There is a little bit possibility that a internal key format might differ from the original one when TTL <= 0 according to the spec

- (int32_t)Timestamp(creation) is suffixed to values in Put internally
- Not specifying/passing or non-positive TTL behaves like TTL = infinity

Depending on how rocksdb implement TTL, each key might be suffixed by Timestamp(creation) regardless of the value of TTL. 

if this is the case then we can't do this with the current PR

- open a file with TTL <= 0
- put some records
- close the file
- open the file with TTL > 0

because the internal key format of records being used during the first open is not suffixed by Timestamp(creation).

and also

> That in case you open using rocksdb::DBWithTTL::Open having TTL <= 0 it's same stuff like using rocksdb::DB::Open where records are never removed based on any TTL.

if this is true then we can rewrite the code like below (no branch by ttl <= 0) to make the code simpler.

```cpp
rocksdb::DBWithTTL* db_with_ttl(0);
status = rocksdb::DBWithTTL::Open(*options, db_name, &db_with_ttl, ttl);
db = db_with_ttl;
```

Reply:
> open_with_ttl/3 and internally to native code I will send a tuple like {use_ttl, TTL}.

Sounds good :)
Please go ahead with this direction.